src/dialog/src/Dialog.js

```diff
@@ -50,28 +50,88 @@ const animationStyles = {
 
 class Dialog extends React.Component {
   static propTypes = {
+    /**
+     * Composes the Overlay component as the base.
+     */
     ...Overlay.propTypes,
+
+    /**
+     * Children can be a node or a function accepting `({ close })`.
+     * See an example to understand how this works.
+     */
+    children: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
```

Comment:
Might be worth making `children` required because a dialog without children is pretty useless. 😛

Reply:
True